core/include/geometry/simple_geometry.hpp

```diff
@@ -0,0 +1,218 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+#include <iterator>
+#include <utility>
+
+#include "core/mask_store.hpp"
+#include "geometry/surface_base.hpp"
+#include "geometry/volume.hpp"
+#include "masks/masks.hpp"
+#include "utils/enumerate.hpp"
+#include "utils/indexing.hpp"
+
+namespace detray {
+/**
+ * @brief Simple geometry implementation
+ *
+ * This class provides a geometry that defines logic volumes which contain
+ * the detector surfaces, joined together by dedicated portal surfaces. It
+ * exports all types needed for navigation and strictly only keeps the
+ * index data (links) that define the geometry relations. The simple geometry
+ * itself makes no distinction between surfaces and portals. Both carry the
+ * same link type: a portal points to the next volume, a surface to the current
+ * volume
+ *
+ * @tparam array_type the type of the internal array, must have STL
+ *                    semantics
+ * @tparam vector_type the type of the internal array, must have STL
+ *                     semantics
+ * @tparam surface_source_link the type of the link to an external surface
+ *                             source
+ *
+ * @note The geometry knows nothing about coordinate systems. This is
+ *       handeled by geometry access objects (e.g. the grid).
+ */
+template <template <typename, unsigned int> class array_type = darray,
+          template <typename> class vector_type = dvector,
+          template <typename...> class tuple_type = dtuple,
+          typename surface_source_link = dindex,
+          typename bounds_source_link = dindex>
+class simple_geometry {
```

Comment:
Can we rename this? `simple` is a judgement, and one might still think this is complicated. For a new naming, think about "what does characterise this geometry" - that should be the name.

Reply:
Is it that portals & surfaces are harmonised?